driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -3070,6 +3080,8 @@ public String toString() {
             case UP:
               Host upHost = metadata.getHost(address);
               if (upHost == null) {
+                // We don't have enough information to create a new Host (we are missing it's ID)
+                // so trigger a full node refresh
                 submitNodeListRefresh();
```

Comment:
This logic changed with the Cloud API, previously we would create a new Host here calling `Metadata.newHost()` but now it's not possible anymore because we don't have the host ID.